storage/execout/file.go

```diff
@@ -2,213 +2,332 @@ package execout
+		}
+		fr.reader = r
+	} else {
+		fr.reader = r
 	}
 
-	return cacheItem.Payload, found
+	return nil
 }
 
-func (c *File) GetAtBlock(blockNumber uint64) ([]byte, bool) {
-	c.Lock()
-	defer c.Unlock()
-
-	if c.deletedBefore != nil && blockNumber < *c.deletedBefore {
-		panic("trying to get deleted data, this should never happen")
+func (fr *fileReader) ReadNext() (*pboutput.Item, error) {
+	if fr.complete {
+		return nil, io.EOF
 	}
-	for _, value := range c.Kv {
-		if value.BlockNum == blockNumber {
-			return value.Payload, true
+	item, err := streamproto.ReadNextItem(fr.reader)
+	if err != nil {
+		if err == io.EOF {
+			fr.complete = true
+			return nil, io.EOF
 		}
+		return nil, err
 	}
-
-	return nil, false
+	fr.lastReadItem = item
+	return item, nil
 }
 
-var MaxExecoutSegmentSize = int(8589934592)
-var ErrSegmentSizeExceeded = connect.NewError(connect.CodeInvalidArgument, fmt.Errorf("execution output segment size exceeded %d bytes: substreams cannot process this segment", MaxExecoutSegmentSize))
-
-func (c *File) Load(ctx context.Context) error {
-	c.Lock()
-	defer c.Unlock()
-	if c.loaded {
-		return nil
+func rewriteAsOrdered(ctx context.Context, r io.Reader, readBytes []byte, store dstore.Store, moduleName, filename string, rng *block.Range, logger *zap.Logger) error {
+	bytes, err := io.ReadAll(r)
+	if err != nil {
+		return fmt.Errorf("reading store file %s: %w", filename, err)
 	}
+	bytes = append(readBytes, bytes...)
 
-	filename := computeDBinFilename(c.Range.StartBlock, c.Range.ExclusiveEndBlock)
-	c.logger.Debug("loading execout file", zap.String("file_name", filename), zap.Object("block_range", c.Range))
-
-	err := derr.RetryContext(ctx, 5, func(ctx context.Context) error {
-		objectReader, err := c.store.OpenObject(ctx, filename)
-		if err == dstore.ErrNotFound {
-			return derr.NewFatalError(err)
-		}
-
-		if err != nil {
-			return fmt.Errorf("loading block reader %s: %w", filename, err)
-		}
-		defer objectReader.Close()
+	o := &pboutput.Array{}
+	if err := o.UnmarshalVTUnsafe(bytes); err != nil {
+		return fmt.Errorf("unmarshalling data: %w", err)
+	}
+	items := o.Items
 
-		// Limit reading to MaxExecoutSegmentSize to prevent excessive memory usage
-		limitedReader := io.LimitReader(objectReader, int64(MaxExecoutSegmentSize+1))
-		bytes, err := io.ReadAll(limitedReader)
-		if err != nil {
-			return fmt.Errorf("reading store file %s: %w", filename, err)
-		}
-		if len(bytes) == MaxExecoutSegmentSize+1 {
-			return derr.NewFatalError(fmt.Errorf("%w: file %s on module %s", ErrSegmentSizeExceeded, filename, c.ModuleName))
-		}
+	sort.Slice(items, func(i, j int) bool {
+		return items[i].BlockNum < items[j].BlockNum
+	})
 
-		outputData := &pboutput.Map{}
-		if err = outputData.UnmarshalFast(bytes); err != nil {
-			return fmt.Errorf("unmarshalling file %s: %w", filename, err)
+	store.SetOverwrite(true)
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel() // closes the WriteAsYouGo
```

Comment:
the `WriteAsYouGo` don't make sense anymore as it's been replaced with something else no?